src/query/generated/proto/prompb/types.proto

```diff
@@ -37,18 +41,29 @@ message LabelMatcher {
     RE  = 2;
     NRE = 3;
   }
-  Type type    = 1;
+  Type type   = 1;
   bytes name  = 2;
   bytes value = 3;
 }
 
-enum Type {
-  GAUGE = 0;
-  COUNTER = 1;
-  TIMER = 2;
+enum MetricType {
+  UNKNOWN         = 0;
+  COUNTER         = 1;
+  GAUGE           = 2;
+  HISTOGRAM       = 3;
+  GAUGE_HISTOGRAM = 4;
+  SUMMARY         = 5;
+  INFO            = 6;
+  STATESET        = 7;
+}
+
+enum M3Type {
```

Comment:
This name was suggested by @shreyassrivatsan when we were discussing how to work around this naming collision. And I agree with it. Otherwise we would have `MetricType` (which comes from Prometheus and is not under our control), `LabelMatcher.Type` and just `Type`, all in the same file. Really want to be more specific here to avoid any confusion.

Reply:
nit: can we at least change the enums from `M3_GAUGE` to `GAUGE`? Otherwise we get janky statements like `prompb.M3Type_M3_GAUGE:   ts.M3MetricTypeGauge`

Reply:
Unfortunately, no. Because of how protobuf works, enum value names must be unique per package, nor just per enum. If I try to make it nice as you suggest, protobuf generation fails with `github.com/m3db/m3/src/query/generated/proto/prompb/types.proto:61:3: Note that enum values use C++ scoping rules, meaning that enum values are siblings of their type, not children of it.  Therefore, "GAUGE" must be unique within "m3prometheus", not just within "M3Type".
`